### PR TITLE
APP-480: [C++] Fix credentials demo

### PR DESCRIPTION
**Pull request by Dubland**
/cc @Slash32

### CODE REVIEW

File: credentials/source/cpp/src/KaaDemo.cpp [hunk 1]
```diff
@@ -45,14 +45,24 @@ class FailoverStrategy: public DefaultFailoverStrategy
     }
 };
 
+class StateListener : public KaaClientStateListener
+{
+    void onConnectionEstablished(const EndpointConnectionInfo& connection)
```
Comment (ohamov) [inline credentials/source/cpp/src/KaaDemo.cpp]: please add `override`

**Review by ohamov — COMMENTED** [review]
A nitpick.

**Review by ohamov — CHANGES_REQUESTED** [review]
I bet it is not compiled now. Check access modifiers inside of `StateListener`.

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Dubland** [comment]
@kaa-leeroy test this please

File: credentials/source/cpp/src/KaaDemo.cpp [hunk 2]
```diff
@@ -45,14 +45,24 @@ class FailoverStrategy: public DefaultFailoverStrategy
     }
 };
 
+class StateListener : public KaaClientStateListener
+{
+    void onConnectionEstablished(const EndpointConnectionInfo& connection) override
```
Comment (ohamov) [inline credentials/source/cpp/src/KaaDemo.cpp:51]: should be `public`

**Comment by Dubland** [comment]
tested with demo application

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Review by Slash32 — APPROVED** [review]

**Pull request merged**